recipes-bsp/u-boot/files/0013-iot2050-add-smbios-and-sysinfo-support.patch

```diff
@@ -0,0 +1,530 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Li Hua Qian <huaqian.li@siemens.com>
+Date: Thu, 20 Jun 2024 15:32:37 +0800
+Subject: [PATCH] iot2050: add smbios and sysinfo support
+
+Signed-off-by: Baocheng Su <baocheng.su@siemens.com>
+Signed-off-by: Li Hua Qian <huaqian.li@siemens.com>
+---
+ .../dts/k3-am65-iot2050-common-u-boot.dtsi    |  18 ++
+ board/siemens/iot2050/board.c                 | 158 ++++++++++++------
+ common/spl/Kconfig                            |   5 +
+ configs/iot2050_defconfig                     |   3 +
+ drivers/sysinfo/Kconfig                       |   7 +
+ drivers/sysinfo/Makefile                      |   1 +
+ drivers/sysinfo/iot2050.c                     | 144 ++++++++++++++++
+ drivers/sysinfo/iot2050.h                     |  26 +++
+ 8 files changed, 308 insertions(+), 54 deletions(-)
+ create mode 100644 drivers/sysinfo/iot2050.c
+ create mode 100644 drivers/sysinfo/iot2050.h
+
+diff --git a/arch/arm/dts/k3-am65-iot2050-common-u-boot.dtsi b/arch/arm/dts/k3-am65-iot2050-common-u-boot.dtsi
+index 082a3c89d0fd..b239b48514a8 100644
+--- a/arch/arm/dts/k3-am65-iot2050-common-u-boot.dtsi
++++ b/arch/arm/dts/k3-am65-iot2050-common-u-boot.dtsi
+@@ -14,6 +14,24 @@
+ 		spi0 = &ospi0;
+ 	};
+ 
++	sysinfo {
++		compatible = "siemens,iot2050-sysinfo";
++		/* TI_SRAM_SCRATCH_BOARD_EEPROM_START */
++		offset = <0x40280000>;
++		bootph-pre-ram;
++
++		smbios {
++			system {
++				manufacturer = "SIEMENS AG";
++				product = "SIMATIC IOT2050";
++			};
++
++			baseboard {
++				manufacturer = "SIEMENS AG";
++			};
++		};
++	};
++
+ 	leds {
+ 		bootph-pre-ram;
+ 		status-led-red {
+diff --git a/board/siemens/iot2050/board.c b/board/siemens/iot2050/board.c
+index 8497212ab890..5ae5e3b3183f 100644
+--- a/board/siemens/iot2050/board.c
++++ b/board/siemens/iot2050/board.c
+@@ -21,32 +21,14 @@
+ #include <spl.h>
+ #include <version.h>
+ #include <linux/delay.h>
++#include <linux/string.h>
+ #include <asm/arch/hardware.h>
+ #include <asm/gpio.h>
+ #include <asm/io.h>
+ 
+-#define IOT2050_INFO_MAGIC		0x20502050
+-
+-struct iot2050_info {
+-	u32 magic;
+-	u16 size;
+-	char name[20 + 1];
+-	char serial[16 + 1];
+-	char mlfb[18 + 1];
+-	char uuid[32 + 1];
+-	char a5e[18 + 1];
+-	u8 mac_addr_cnt;
+-	u8 mac_addr[8][ARP_HLEN];
+-	char seboot_version[40 + 1];
+-	u8 padding[3];
+-	u32 ddr_size_mb;
+-} __packed;
++#include "../../../../drivers/sysinfo/iot2050.h"
```

Comment:
This looks not so nice. Not sure if this relative include deep into the driver is acceptable. I rather suspect the header needs to move to include/.